pkg/hostagent/services.go

```diff
@@ -618,9 +621,7 @@ func (sep *serviceEndpoint) SetOpflexService(ofas *opflexService, as *v1.Service
 			} else {
 				sm.ServiceIp = as.Spec.ClusterIP
 			}
-			if as.Spec.SessionAffinityConfig != nil && as.Spec.SessionAffinity == "ClientIP" {
-				sm.SessionAffinity = getSessionAffinity(as.Spec.SessionAffinityConfig)
-			}
+			sm.SessionAffinity = getSessionAffinity(as)
```

Comment:
is it not fine as the way it is earlier that ---> if check first and then function call rather than doing it inside the function. In the older way, we can save the function call and push pop of calls on the program stack

Reply:
No, here we are always creating a 1Sec timer. so no need to have a call

Reply:
As discussed, the comment was minor and for generic refactoring code as if needed. The comment was majorly focused to save a function call. But the bigger use case was for modularising it in a single place which takes the higher priority. Aprooving